small-talk/components/smolAuth.js

```diff
@@ -0,0 +1,55 @@
+import { getCookie } from "@/util/smolCookie";
+
+export function smolAuth(getServerSidePropsFunc) {
+    return async (ctx) => {
+        const { req } = ctx;
+        
+        if (req.headers.cookie) {
+            const session_id = getCookie("session_id", req.headers.cookie);
+            //console.log(session_id)
+            try {
+                let response = await fetch("http://" + req.headers.host + "/api/auth2/", {
+                    method: "POST",
+                    headers: {
+                        "Content-Type" : "application/json",
+                        "Authorization" : session_id
+                    }
+                })
+
+                let json = (await response.json());
+                
+                if (json == null) {
+                    return {
+                        redirect: {
+                        permanent: false,
+                        destination: '/login',
+                        },
+                    };
+                }
+                console.log(json)
+                let isFound = (json.found != null) ? json.found : false;
+
+                if (isFound == false) {
+                    return {
+                        redirect: {
+                        permanent: false,
+                        destination: '/login',
+                        },
+                    };
+                }
+            } catch (error) {
+                // Failure in the query or any error should fallback here
+                // this route is possibly forbidden means the cookie is invalid
+                // or the cookie is expired
+                return {
+                    redirect: {
+                        permanent: false,
+                        destination: '/login',
+                    },
```

Comment:
Ah okay I see if there's any error in the api call, then it would default here. So error handling isn't exactly necessary in the auth2.js? Might be helpful to add some additional logs in there to track down any potential errors

Reply:
This is a good suggestion. We should implement this in the next iteration